src/NUnitEngine/nunit.engine.core/Runners/AggregatingTestRunner.cs

```diff
@@ -55,7 +55,8 @@ public IList<ITestEngineRunner> Runners
                 if (_runners == null)
                 {
                     _runners = new List<ITestEngineRunner>();
-                    foreach (var subPackage in TestPackage.SubPackages)
+                    foreach (var subPackage in TestPackage.Select(p => !p.HasSubPackages()))
```

Comment:
I'm confused by this expression. It looks like we're selecting a `bool` value, meaning that `subPackage` is a boolean. I'm guessing this is not the System.Linq `Select` but rather something more like System.Linq's `Where` with ` `.SubPackages.Where`?

Reply:
It's an extension I wrote to TestPackage a while back, which selects packages recursively out of the tree. IIRC I used Select to avoid confusion (with Linq Where) but I guess I didn't succeed. :-) It's used in a number of places.